lib/internal/modules/package_json_reader.js

```diff
@@ -13,29 +13,33 @@ let manifest;
  *
  * @param {string} jsonPath
  */
-function read(jsonPath) {
-  if (cache.has(jsonPath)) {
-    return cache.get(jsonPath);
-  }
+ const validators = require('internal/validators');
 
-  const { 0: string, 1: containsKeys } = internalModuleReadJSON(
-    toNamespacedPath(jsonPath)
-  );
-  const result = { string, containsKeys };
-  const { getOptionValue } = require('internal/options');
-  if (string !== undefined) {
-    if (manifest === undefined) {
-      manifest = getOptionValue('--experimental-policy') ?
-        require('internal/process/policy').manifest :
-        null;
-    }
-    if (manifest !== null) {
-      const jsonURL = pathToFileURL(jsonPath);
-      manifest.assertIntegrity(jsonURL, string);
-    }
-  }
-  cache.set(jsonPath, result);
-  return result;
-}
+ function read(jsonPath) {
+   if (cache.has(jsonPath)) {
+     return cache.get(jsonPath);
+   }
+ 
+   const { 0: string, 1: containsKeys } = internalModuleReadJSON(
+     toNamespacedPath(jsonPath)
+   );
+   const result = { string, containsKeys };
+   const { getOptionValue } = require('internal/options');
+   if (string !== undefined) {
+     validators.validateString(string, 'string');
```

Comment:
Take a look at https://github.com/nodejs/node/blob/de696d70ef1d0a5ff52312709914134306bd7c82/src/node_file.cc#L987-L1081. `string` is always guaranteed to be either a string or undefined, so this validation is not required.

Also, these validators are used only for validating user input and this is not user input - it's the contents of the read `package.json` file.

Reply:
yes thank you very much I just realized it but wouldn't it be nice to verify if it's a string :/

Or does the area you mention perform this function? @RaisinTen

Reply:
> thank you very much

Any time :) 

> but wouldn't it be nice to verify if it's a string :/

Since it's guaranteed to be a string / undefined, this validation would add more code to the module loader, which is a very hot code path. This would unfortunately slow things down without any benefit.

> Or does the area you mention perform this function?

It doesn't perform a validation because the binding either returns an empty array or one that contains [string, boolean], so the validation isn't necessary in this case because user code cannot possibly make it return anything apart from these values

Reply:
I understood, thank you very much again, I would like to contribute node, I would be very grateful if you have any advice for me to start @RaisinTen

Reply:
That's great to know! DM'ing you about it :+1: